src/entities.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import json
+import re
+
+### Declare headers
+header_a = """/*
+ * Generated file - do not edit directly.
+ * This file was generated from:
+ *     https://html.spec.whatwg.org/entities.json
+ * by means of entities.py in the libgd src dir
+*/
+"""
+
+header_b = """
+#include "entities.h"
+
+const struct entities_s entities[NR_OF_ENTITIES] = {"""
+
+header_c = """/*
+ * Generated file - do not edit directly.
+ * This file was generated from:
+ *     https://html.spec.whatwg.org/entities.json
+ * by means of entities.py in the libgd src dir
+*/
+#ifndef GD_ENTITIES_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+//html entity strings are entity prefix + string + suffix limited
+//hex and dec should be limited to current unicode spec + entity prefix + suffix
+"""
+
+header_d = """
+struct entities_s{
+                  	char *name;
+                        uint32_t unicode_a;
+                        uint32_t unicode_b;
+                };
+extern const struct entities_s entities[NR_OF_ENTITIES];
+
+#endif
+"""
+
+### Open files
+file_json = open("entities.json", mode="r")
+file_ent_h = open("entities.h", mode="w")
+file_ent_c = open("entities.c", mode="w")
+
+### Initialize some objects
+x = json.load(file_json)
+curline = 1
+totallines = 0
+len_name_max = 0
+p = re.compile("&\S+;")
+
+### Print file headers
+file_ent_c.write(header_a + header_b + "\n")
+file_ent_h.write(header_c)
+file_ent_h.write("#define GD_ENTITIES_H 1\n")
+
+### Count how many matching lines we have
+for e in x:
```

Comment:
i think this for loop can be one line:
```
totallines = sum(p.match(x) for x in entities)
```

then again, we can delete this loop entirely -- see below